alf/environments/parallel_environment.cpp

```diff
@@ -95,12 +95,25 @@ class SharedDataBuffer {
   std::vector<size_t> offsets_;  // offset of the i-th array
   std::vector<size_t> sizes_;    // size of each slice of the i-th array
   std::vector<py::buffer_info> buffer_infos_;
+  std::vector<std::vector<ssize_t>> slice_strides_;
 
   inline char* GetBuf(int slice_id, int array_id) const {
     return buf_ + offsets_[array_id] + sizes_[array_id] * slice_id;
   }
 };
 
+void CheckStrides(const py::buffer_info& info,
+                  const std::vector<ssize_t>& strides) {
+  if (info.strides != strides) {
+    throw std::runtime_error(
+        py::str(
+            "strides mismatch. Expected: {}, "
+            "Got: {}, shape: {}. It might be caused by non-continguous array, "
+            "which can be fixed by copying the array.")
+            .format(strides, info.strides, info.shape));
```

Comment:
Is it possible to also print out more buffer info? Like buffer's name. Just looking at the shape might still not be clear.

Reply:
buffer_info does not have name